File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    sq_dist = x ** 2 + y ** 2
    cond do
      sq_dist <= 1 -> 10
      sq_dist <= 25 -> 5
      sq_dist <= 100 -> 1
      true -> 0
    end
  end
end

File: python/darts/darts.py
"""Calculate the points scored in a single toss of a Darts game"""

def score(x_dist, y_dist):
    square_dist = x_dist ** 2 + y_dist ** 2
    return (
        10 if square_dist <= 1 ** 2 else
        5 if square_dist <= 5 ** 2 else
        1 if square_dist <= 10 ** 2 else
        0
    )
